specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+        }
+      },
+      "description": "Request from client to check resource name availability."
+    },
+    "NameAvailability": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "Error Message."
+        },
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "Indicates whether the resource name is available."
+        },
+        "reason": {
+          "type": "string",
+          "description": "Reason for name being unavailable."
+        }
+      },
+      "description": "Represents a resource name availability."
+    },
+    "Identity": {
+      "x-ms-client-flatten": true,
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "type": "string",
+          "description": "ObjectId from the KeyVault",
+          "readOnly": true
+        },
+        "tenantId": {
+          "type": "string",
+          "description": "TenantId from the KeyVault",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of managed service identity.",
+          "enum": [
+            "UserAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "ManagedServiceIdentityType",
+            "modelAsString": false
+          }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "object",
+            "description": "",
+            "items": {
+              "$ref": "#/definitions/UserAssignedIdentity"
+            }
+          },
+          "description": "Metadata of user assigned identity."
+        }
+      },
+      "description": "Properties to configure Identity for Bring your Own Keys"
+    },
+    "UserAssignedIdentity": {
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "Principal Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "PrincipalId",
+          "readOnly": true
+        },
+        "clientId": {
+          "description": "Client Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "ClientId",
+          "readOnly": true
+        }
+      },
+      "description": "Metadata of user assigned identity."
+    },
+    "CutoverMigrationRequest": {
+      "type": "object",
+      "description": "Cutover migration request",
+      "properties": {
+        "dnsChangeMode": {
+          "type": "string",
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
```

Comment:
>sourceServerResourceId [](http://example.com/codeflow?start=9&length=22)

The resource should be in the URI path rather than the body. POST actions on single resources should just be the resource path + the action name. In this case you wouldn't have a request payload.

Reply:
Here the API is hosted on Flexible Server RP.
And the sourceServerResourceId represent a single server resource Id.

Reply:
OK, but I think this action should be performed on a specific flexible server (the same URI path that cutoverMigration is on). Then you don't have sourceServerResourceId in the body because it's already in the URI.

Reply:
At the time of the assessment, the target server is not present.
The user first calls the assessForMigration api for their single server to check if the migration to a target flexible server would be successful or if there are any blockers.

And the reason for us to host the API on Target(Flexible) RP and not the Source(Single) RP is as below:
1.Single Server code base is a dying code base. In some months from now, we will not allow any more customers to create a single server
2.The code base is complex and hard to test any new changes right now
3.The train deployment takes months to roll out , and making any iterative changes would mean a long process for us to go to public preview and GA

And if the question is that, how are we getting the information about a single server from a Flexible RP,
Thankfully, we already have CAS endpoints(Internal apis) present on Single Server RP, and cross platform communication also available.
We just call the Single Server CAS endpoint from FLexible RP and get the information we need.

Reply:
I see: you specifically don't want the request routed on the server being migrated, so you put it at tenant level and pass in its ID. Given the situation, I think it makes sense.